Anforderungsanalyse (#466)
project-leek/project-leek: docs/report/report.tex

Im weiteren Gespräch mit den Kunden ergeben sich weitere ergänzende Anforderungen an das Produkt:
\begin{center}
  \begin{tabular}[h]{c|c}
    Anforderung & Klassifikation \\
    \hline
    Sound Wiedergabe auf der Box selbst & funktional  \\
    Unterstützung anderer Musikdienste & funktional \\
    3D-Modellierung und Print einer passenden Box & nicht funkt.  \\
    Maximaler Kostenpunkt von 30€ & nicht funkt.  \\
    Cloud-Anbindung der Box & funktional \\
    Auslieferung des UI aus der Cloud & funktional \\
    \end{tabular}
\end{center}

Durch die agile Entwicklung des Projektes sind diese Anforderungen iterativ von Sprint zu Sprint gewachsen und neue implizite Anforderungen sind aus den bestehenden Anforderungen entstanden.
Um den Kostenpunkt von maximal 30€ zu realisieren entstand beispielsweise die Idee, einer kleinen kostengünstigeren Version, bei der die Kosten weniger als 30€ betragen und nur die Grundfunktionen umsetzt, und einer größerer umfangreicheren, die weitere Funktionen beinhaltet wie das Abspielen der Musik über einen eigenen Lautsprecher.
Das Team stieß auch während der Entwicklung auf wiedersprechende Anforderungen.

> So wurde sich zum Beispiel gegen bezahlte Services \colorbox{red}{für subdomains (bessere formulierung?)} entschieden, was zwar die Usability etwas einschränkte, jedoch mehr dem Open-Source-Gedanken des Projektes entspricht.

Reviewer: Hier kann eventuell @anbraten bei der Formulierung unterstützen :)

> So wurde sich zum Beispiel gegen bezahlte Services \colorbox{red}{für subdomains (bessere formulierung?)} entschieden, was zwar die Usability etwas einschränkte, jedoch mehr dem Open-Source-Gedanken des Projektes entspricht.

Reviewer 2: ```suggestion
So wurde sich zum Beispiel gegen bezahlte Services \colorbox{red}{für subdomains (bessere formulierung?)} entschieden, was die Usability etwas einschränkte, jedoch mehr dem Open-Source-Gedanken des Projektes entspricht.
```

uf, "was die... einschränke" klingt nicht gut, aber mir fällt grade nichts besseres ein.

\subsection{Unternehmen}
\begin{quote}
  \textit{Ich denke, dass jede allgemein nützliche Information frei sein sollte.
  Mit 'frei' beziehe ich mich nicht auf den Preis, sondern auf die Freiheit, Informationen zu kopieren und für die eigenen Zwecke anpassen zu können.

>   Wenn Informationen allgemein nützlich sind, wird die Menschheit durch ihre Verbreitung reicher, ganz egal, wer sie weiter gibt und wer sie erhält.}

Reviewer: Das ist doch ein Zitat. Da würde ich glaube ich nichts drin ändern. Es sei denn Julian hat falsch Zitiert

Author: hab dort nur im der tex zeilenumbrüche reingepackt, um das in der tex besser lesbarer zu haben

Reviewer: Ahh ok :D

\subsection{Produktvision}
Das Unternehmen macio stellte in ihrer Ausschreibung als einzige Firma die Anforderung, dass Projekt quelloffen (\textit{Open Source}) umzusetzen.
Damit zeigten sie die Bereitschaft, die vom Team geleistete Arbeitszeit nicht für den Eigenbedarf zu nutzen, sondern einen Beitrag für die Open-Source-Community zu leisten.
Ihr Hauptgeschäft ist das Design und Entwickeln von Mensch-Maschine-Interfaces.
Als etabliertes Unternehmen konnten Sie neben der Rolle des Kunden auch ihre jahrelange Erfahrung bei Bedarf einbringen und Hilfestellung leisten.
So gab beispielsweise ein Designer des Unternehmens Feedback zum Entwurf der Benutzeroberfläche
\subsection{Anforderungsanalyse}
Im Rahmen des Projekt Informatik möchte Macio ihr Portfolio im IoT-Bereich erweitern, sowie ihren Empfangsraum im Standort Kiel verschönern.
Hierfür soll eine smarte Spielzeug-Box gebaut werden.

> Smarte Spielzeuge gibt es im kommerziellen Bereich viele, daher soll dieses Projekt eine Open-Source-Alternative schaffen.\\

Reviewer: Ich finde es wird nicht richtig klar, dass es keine Open-Source Alternativen gibt. 

Vielleicht eher. Es gibt viele smarte Spielzeuge ABER alle komerziell. Deswegen dieses Projekt, damit es ein Open-Soruce Spielzeug gibt

Author: umgesetzt in https://github.com/project-leek/project-leek/pull/466/commits/bac512992bc6f3232f89c898a0d23eebfb63243b

> Mithilfe der von macio gestellten Projektskizze (siehe \ref{pdf:macioprojektskizze}) haben wir uns folgende Produktvision vorgestellt:

Reviewer: ```suggestion
Mithilfe der von macio gestellten Projektskizze (siehe \ref{pdf:macioprojektskizze}) haben erstellte das Team die folgende Produktvision:
```

Author: umgesetzt in https://github.com/project-leek/project-leek/pull/466/commits/bac512992bc6f3232f89c898a0d23eebfb63243b

> Das Produkt ist eine Musik-Box, die NFC-Chips lesen und Spotify Connect unterstützen soll.

Reviewer: ```suggestion
Angestrebt wird die Konzeption und Entwicklung einer Musik-Box, die NFC Tags lesen kann und Schnittstellen zu Musik-Streaming-Anbiertern (wie Spotify) besitzt.
```

Author: umgesetzt in https://github.com/project-leek/project-leek/pull/466/commits/bac512992bc6f3232f89c898a0d23eebfb63243b

> Auf die Box können dann Spielzeuge (z.B. in Form von kleinen Figuren) mit integrierten NFC-Chips gestellt werden, um spezifische Musik abspielen zu lassen.
> Die Musik wird von Spotify-Connect über eine bereits bestehende Musik-Anlage abgespielt.

Reviewer: ```suggestion
Wird auf die Box beispielsweise ein Spielzeuge (z.B. in Form von kleinen Figuren) mit integrierten NFC-Chips gestellt soll ein frei konfigurierbares Musikstück über eine bereits bestehende Musik-Anlage abgespielt abgespielt werden.
```

Author: umgesetzt in https://github.com/project-leek/project-leek/pull/466/commits/bac512992bc6f3232f89c898a0d23eebfb63243b

Falls es im Rahmen des Projektes möglich ist, sollen die Nutzer in der Lage sein, zwischen verschiedenen Musikanbietern zu wechseln.
NFC-Chips und die zugehörige Musik sollen über ein Web-basierte Benutzeroberfläche konfiguriert werden können.
Diese Benutzeroberfläche soll von der Box ausgeliefert und primär für Smartphone-Bedienung gestaltet werden.
Da es sich um ein Open Source Projekt mit entsprechender Lizenz handelt, muss auch eine aussagekräftige, öffentliche Dokumentation verfasst werden.
Macio stellt die benötigte Hardware zur Verfügung und unterstützt bei technischen Fragen.\\

\subsubsection{Anforderungen}

> Aus der Projektskizze und der Produktvision lassen sich konkreter folgende Mindestanforderungen und entsprechende Klassifikationen an das Produkt ableiten:

Reviewer: ```suggestion
Aus der Projektskizze und der Produktvision wurden folgende konkrete Mindestanforderungen und entsprechende Klassifikationen an das Produkt abgeleitet:
```

Author: umgesetzt in https://github.com/project-leek/project-leek/pull/466/commits/bac512992bc6f3232f89c898a0d23eebfb63243b

\begin{center}
  \begin{tabular}[h]{c|c}
    Anforderung & Klassifikation \\
    \hline
    NFC-Tags lesen, schreiben und entschlüsseln & funktional  \\
    Mit Spotify Connect verbinden und arbeiten & funktional \\

>     Responsive UI konzeptionieren und umsetzen & funktional  \\

Reviewer: Was heißt dieses "& funktional"?

Author: das ist ne tabelle, das & trennt die spalten

Author: umgesetzt in https://github.com/project-leek/project-leek/pull/466/commits/bac512992bc6f3232f89c898a0d23eebfb63243b

    Aussagekräftige Dokumentation mit Benutzerhandbuch & nicht funkt.  \\
  \end{tabular}
\end{center}

> Um diese Anforderungen weiter zu definieren hat das Team hierfür User Stories erstellt.
> Durch die User Stories wurde sichergestellt, dass beide Parteien (der Kunde und die Entwickler) die Anforderungen auf gleiche Weise interpretieren und verstehen.

Reviewer: ```suggestion
Um diese Anforderungen weiter zu definieren erstellte das Team User Stories,
durch die sichergestellt wurde,  dass beide Parteien (der Kunde und die Entwickler) die Anforderungen  gleich interpretieren und verstanden.
```

Author: umgesetzt in https://github.com/project-leek/project-leek/pull/466/commits/bac512992bc6f3232f89c898a0d23eebfb63243b

> Dabei entstand auch ein Priorisierung der Anforderungen bemessen an dem Return of Investment und dem Nutzen für den Endnutzer.

Reviewer: ```suggestion
Dabei entstand auch eine Priorisierung der Anforderungen bemessen an dem \textit{Return of Investment} und dem Nutzen für den Endnutzer.
```

Author: umgesetzt in https://github.com/project-leek/project-leek/pull/466/commits/bac512992bc6f3232f89c898a0d23eebfb63243b

Im weiteren Gespräch mit den Kunden ergeben sich weitere ergänzende Anforderungen an das Produkt:
\begin{center}
  \begin{tabular}[h]{c|c}
    Anforderung & Klassifikation \\
    \hline
    Sound Wiedergabe auf der Box selbst & funktional  \\
    Unterstützung anderer Musikdienste & funktional \\
    3D-Modellierung und Print einer passenden Box & nicht funkt.  \\
    Maximaler Kostenpunkt von 30€ & nicht funkt.  \\
    Cloud-Anbindung der Box & funktional \\
    Auslieferung des UI aus der Cloud & funktional \\
    \end{tabular}
\end{center}

> Durch die agile Entwicklung des Projektes sind diese Anforderungen iterativ von Sprint zu Sprint gewachsen.
> Neue implizite Anforderungen sind aus den bestehenden Anforderungen entstanden.

Reviewer: ```suggestion
Durch die agile Entwicklung des Projektes sind diese Anforderungen iterativ von Sprint zu Sprint gewachsen und
neue implizite Anforderungen sind aus den bestehenden Anforderungen entstanden.
```

Author: umgesetzt in https://github.com/project-leek/project-leek/pull/466/commits/bac512992bc6f3232f89c898a0d23eebfb63243b

> Beispielsweise um den Kostenpunkt von maximal 30€ zu realisieren entstand die Idee einer kleinen kostengünstigeren Version, welche unter 30€ liegt und nur die Grundfunktionen umsetzt, und einer größerer umfangreicheren, die weitere Funktionen beinhaltet wie das Abspielen der Musik über einen eigenen Lautsprecher.

Reviewer: ```suggestion 
Um den Kostenpunkt von maximal 30€ zu realisieren entstand beispielsweise die Idee, einer kleinen kostengünstigeren Version, bei der die Kosten weniger als 30€ betragen und nur die Grundfunktionen umsetzt, und einer größerer umfangreicheren, die weitere Funktionen beinhaltet wie das Abspielen der Musik über einen eigenen Lautsprecher.
```

Author: umgesetzt in https://github.com/project-leek/project-leek/pull/466/commits/bac512992bc6f3232f89c898a0d23eebfb63243b

Das Team stieß auch während der Entwicklung auf wiedersprechende Anforderungen.

> So wurde sich zum Beispiel gegen bezahlte Services \colorbox{red}{für subdomains (bessere formulierung?)} entschieden auf Kosten der Usability, da dies mehr der Opensource Natur des Projektes entspräche.

Reviewer: ```suggestion
So wurde sich zum Beispiel gegen bezahlte Services \colorbox{red}{für subdomains (bessere formulierung?)} entschieden, was zwar die Usability etwas einschränkte, jedoch mehr dem Open-Source-Gedanken des Projektes entspricht.
```

Author: umgesetzt in https://github.com/project-leek/project-leek/pull/466/commits/bac512992bc6f3232f89c898a0d23eebfb63243b

> Weiter wurden auch die Anforderungen über die Entwicklung hinweg durch das Kundenfeedback am Ende der Sprints immer spezifischer.

Reviewer: Was meinst du damit. Wir mir nicht ganz klar :)

Author: hm wird gestrichen, der Satz sollte textfülller sein, aber ist unnötig

Author: umgesetzt in https://github.com/project-leek/project-leek/pull/466/commits/bac512992bc6f3232f89c898a0d23eebfb63243b

Im weiteren Gespräch mit den Kunden ergaben sich weitere ergänzende Anforderungen an das Produkt:
\begin{center}
  \begin{tabular}[h]{c|c}
    Anforderung & Klassifikation \\
    \hline
    Sound Wiedergabe auf der Box selbst & funktional  \\
    Unterstützung anderer Musikdienste & funktional \\
    3D-Modellierung und Print einer passenden Box & nicht funkt.  \\
    Maximaler Kostenpunkt von 30€ & nicht funkt.  \\
    Cloud-Anbindung der Box & funktional \\
    Auslieferung des UI aus der Cloud & funktional \\
    \end{tabular}
\end{center}

Durch die agile Entwicklung des Projektes sind diese Anforderungen iterativ von Sprint zu Sprint gewachsen und neue implizite Anforderungen sind aus den bestehenden Anforderungen entstanden.
Um die Kosten von maximal 30€ zu realisieren entstand beispielsweise die Idee, einer kleinen kostengünstigeren Version, bei der die Kosten weniger als 30€ betragen und nur die Grundfunktionen umsetzt, und einer größerer umfangreicheren, die weitere Funktionen beinhaltet wie das Abspielen der Musik über einen eigenen Lautsprecher.

> Das Team stieß auch während der Entwicklung auf wiedersprechende Anforderungen.
> So wurde sich zum Beispiel gegen bezahlte Services \colorbox{red}{für subdomains (bessere formulierung?)} entschieden, was die Usability etwas einschränkte, jedoch mehr dem Open-Source-Gedanken des Projektes entspricht.

Reviewer 2: ```suggestion
Einige der Anforderungen haben sich während der Entwicklungsphase als widersprüchlich herausgestellt. 
So wurde sich zum Beispiel gegen bezahlte Services entschieden. Dies schränkte die Usability ein, entsprach jedoch mehr dem Open-Source-Gedanken des Projekts.
```
